app/src/main/java/chat/rocket/android/chatroom/ui/bottomsheet/WebUrlBottomSheet.kt

```diff
@@ -0,0 +1,105 @@
+private const val BUNDLE_BOTTOMSHEET_STATE = "bottomsheet_state"
+
+class WebUrlBottomSheet : BottomSheetDialogFragment() {
+
+    private lateinit var webPageUrl: String
+    private lateinit var chatRoomId: String
+    private lateinit var bottomsheetState: String
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        arguments?.run {
+            webPageUrl = getString(BUNDLE_WEB_PAGE_URL, "")
+            chatRoomId = getString(BUNDLE_CHATROOM_ID, "")
+            bottomsheetState = getString(BUNDLE_BOTTOMSHEET_STATE, "")
+        } ?: requireNotNull(arguments) { "no arguments supplied when the fragment was instantiated" }
+        return inflater.inflate(R.layout.webview_bottomsheet, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        setupToolbar()
+        setupWebView()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        var bottomsheet: FrameLayout = dialog.findViewById(com.google.android.material.R.id.design_bottom_sheet)
+        var behaviour = BottomSheetBehavior.from(bottomsheet)
+        behaviour.setBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
+            override fun onSlide(p0: View, p1: Float) {
+            }
+
+            override fun onStateChanged(p0: View, newState: Int) {
+                if (newState == BottomSheetBehavior.STATE_HIDDEN) {
+                    dialog.cancel()
+                }
+                if (newState == BottomSheetBehavior.STATE_DRAGGING) {
+                    behaviour.state = BottomSheetBehavior.STATE_COLLAPSED
+                }
+            }
+
+        })
+        if (bottomsheetState == "tall"){
+            bottomsheet.layoutParams.height = ViewGroup.LayoutParams.MATCH_PARENT
+            behaviour.state = BottomSheetBehavior.STATE_COLLAPSED
+            behaviour.isHideable = false
+        }
+    }
+
+    private fun setupToolbar() {
+        toolbar.title = webPageUrl.replace("https://","")
+        toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+        toolbar.setNavigationOnClickListener {
+            dialog.cancel()
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setupWebView()
+    }
+
+    private fun setupWebView() {
+        with(web_view.settings) {
+            javaScriptEnabled = true
+            domStorageEnabled = true
+        }
+
+        web_view.webViewClient = object : WebViewClient() {
+            override fun onPageFinished(view: WebView, url: String) {
+                super.onPageFinished(view, url)
+                ui {
+                    view_loading.hide()
+                }
+            }
+        }
+        web_view.loadUrl(webPageUrl)
+    }
+
+    companion object {
+        fun newInstance(webPageUrl: String, chatRoomId: String, bottomsheetState: String) = WebUrlBottomSheet().apply {
```

Comment:
You could declare it at the top of the class like we do. This way you can avoid declaring a companion object.